### ISSUE TITLE
Azure Application Risky Perms

### ISSUE THREAD

**Issue by dotnvo** [mandiant/Mandiant-Azure-AD-Investigator#16]
First of all, this looks like a pretty cool little tool - appreciate the work you all have been doing on it.

This is maybe less of an 'issue' but more to understand the guiding principals of decisions made for identifying risky perms when querying applications. I'm taking a look at the JSON config, specifically the Applications and ServicePrincipals node. From reviewing the code these are the API perms you've identified as 'risky'. There are a ton of perms obviously and I think there are plenty of perms that are on the list that are 'obvious'.  However, could you share the process or how you decided which perms are risky and not risky? For Example, I  know one of the applications our team has created is using the following API Permission in conjunction with a SP role, but I do not find that in the JSON. I sort of expected to find it though. The code that would add this API perm in the app manifest looks like this:

```json
"requiredResourceAccess": [
   {
      "resourceAppId": "00000002-0000-0ff1-ce00-000000000000",
      "resourceAccess": [
         {
            "id": "dc50a0fb-09a3-484d-be87-e023b12c6440",
            "type": "Role"
         }
      ]
   }
],
```
This is a JSON rep of the Exchange Manage As App permission for Office 365 Exchange Online.  If I recall correctly it has to be coupled with an SP added to an Exchange role to be functional. I was surprised to not see that in there, but perhaps this is because it does take another thing to make it functional. Here's a SS of the perm:

![image](https://user-images.githubusercontent.com/29506123/186719466-fa980852-3f21-4efd-b65f-e2a7cbb76c6f.png)

That said, I'd personal consider this something worthy of reporting on, even if it requires another piece.  What other permissions have you considered or is that list of risky perms really still a work in progress as you find others that you find worthy of reporting on? Are there certain guiding principals you used to add or not add perms to that list?

**Comment by dotnvo** [comment]
Doesn't look like issues is monitored for this repo so going to close.

**Comment by dmb2168** [comment]
Hi @dotnvo thanks for the feedback!
The initial motivation for "risky permissions" were permissions that could allow a threat actor to persist in a M365 tenant and/or access data at an organizational scale. That being said, Microsoft changes things pretty quickly and I'm sure there are new permissions that we missed.
For exchange.manageasapp the reasoning was that this role is not in and of itself dangerous. You need to also assign the SP an Exchange RBAC role for it to become dangerous. It might be valuable to enumerate SPs that have RBAC roles assigned. I don't think the module does that currently.

**Comment by dotnvo** [comment]
@dmb2168  yeah I noted as much in my initial post and figured that might be the case. It's good to understand the context behind what is considered a risky perm.
